baseplate/integration/mixins/lazy_attributes.py

```diff
@@ -0,0 +1,22 @@
+class LazyAttributesMixin(object):
```

Comment:
I'm looking for some advice on improving this mixin pattern. I'm not a fan of how this relies on the implementing classes to set `self._lazy_attributes = {}`. I _could_ remove that from the implementing classes by modifying `__getattr__` to also:
```
if name == '_lazy_attributes':
  _lazy_attributes = {}
  setattr(self, name, _lazy_attributes)
  return _lazy_attributes
```

This feels gross though.

Reply:
ha, that's super meta.

one thing would just be to do the initiation in a constructor and force callers to call the constructor? that at least hides the details of what's going on from consumers?